website/docs/r/vm_affinity_rule.html.markdown

```diff
@@ -0,0 +1,125 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_vm_affinity_rule"
+sidebar_current: "docs-vcd-resource-vm-affinity-rule"
+description: |-
+  Provides a vCloud Director VM affinity rule resource. This can be
+  used to create, modify, and delete VM affinity and anti-affinity rules.
+---
+
+# vcd\_vm\_affinity\_rule
+
+Provides a vCloud Director VM affinity rule resource. This can be
+used to create, modify, and delete VM affinity and anti-affinity rules.
+
+Supported in provider *v2.9+*
+
```

Comment:
We need a note here about what prerequisites VCD should meet for Affinity Rules to work at all. For example, I'm getting this when trying:
```
vcd_vm_affinity_rule.demo_affinity_rule: Creating...

Error: task did not complete successfully:  [500:INTERNAL_SERVER_ERROR] - [ 20415e53-4f4d-4b8b-bf85-44dfab0a380b ] Internal Server Error

  on yorktown.tf line 561, in resource "vcd_vm_affinity_rule" "demo_affinity_rule":
 561: resource "vcd_vm_affinity_rule" "demo_affinity_rule" {
```

Reply:
I doubt that an internal server error is a result of unmet prerequisites.
The only prerequisites are:
* the VMs exist
* they are not used in another affinity rule

Reply:
Yes, for some reason VMs from that vApp get "refused". Same error in UI. Works with VMs from a different vApp.